### PR TITLE
not blocking render with docsearch assets

### PR DESCRIPTION
**Pull request by yeion7**
Load defer docsearch script and preload stylesheet for #6107

### CODE REVIEW

File: www/src/html.js [hunk 1]
```diff
@@ -63,6 +63,7 @@ export default class HTML extends React.Component {
           />
           {this.props.postBodyComponents}
           <link
+            rel="preload"
```
Comment (jlengstorf) [inline www/src/html.js]: This won't quite work as expected.

With `preload`, we have to declare each thing twice:

1. As a link to be preloaded
2. As the actual resource

Check out this [write-up on preload](https://developer.mozilla.org/en-US/docs/Web/HTML/Preloading_content) for details on how it works.

File: www/src/html.js [hunk 2]
```diff
@@ -54,7 +54,7 @@ export default class HTML extends React.Component {
             color={colors.gatsby}
           />
           <meta name="msapplication-config" content={`/browserconfig.xml`} />
-          <script src="https://cdn.jsdelivr.net/npm/docsearch.js@2/dist/cdn/docsearch.min.js" />
+          <script src="https://cdn.jsdelivr.net/npm/docsearch.js@2/dist/cdn/docsearch.min.js" defer />
```
Comment (jlengstorf) [inline www/src/html.js]: Perfect!

File: www/src/html.js (hunk 1 above)
Comment (jlengstorf) [reply www/src/html.js]: Also, check out [this gist](https://gist.github.com/jlengstorf/4b39f2e5442620f1d26c14d48b580684) for an example of dynamically loading CSS.

**Comment by yeion7** [comment]
@jlengstorf thanks, i maked the change

File: www/src/html.js [hunk 3]
```diff
@@ -64,6 +64,10 @@ export default class HTML extends React.Component {
           {this.props.postBodyComponents}
           <link
             rel="preload"
+            href="https://cdn.jsdelivr.net/npm/docsearch.js@2/dist/cdn/docsearch.min.css"
+            as="style"
+          />
```
Comment (jlengstorf) [inline www/src/html.js]: Almost there! This `preload` link (and only this one) should go right after the opening `<head>` tag.

File: www/src/html.js (hunk 1 above)
Comment (gaui) [reply www/src/html.js]: How about...

```
<link rel="preload" as="style" href="style.css" onload="this.rel='stylesheet'" />
```

**Comment by yeion7** [comment]
Done

**Comment by yeion7** [comment]
Thanks you for help and guide 💚

File: www/src/html.js (hunk 1 above)
Comment (jlengstorf) [reply www/src/html.js]: @gaui I'm not familiar with that approach. What's the goal of the `onload` attribute here?

**Comment by gaui** [comment]
Great job guys and your immediate reaction tells me Gatsby is a quality product.

**Pull request merged**